### ISSUE TITLE
Ability to use non-http-based liveness and readiness probes in the Seldon executor

### ISSUE THREAD

**Issue by austince** [SeldonIO/seldon-core#2238]
Hi there, I'm trying to integrate some SeldonDeployments with the Kuma service mesh and running into a few issues. The main issue is the Kuma doesn't support http-based probes at the moment. We've changed our containers to use an exec-based probe as a workaround, which solves the issue. The only place we cannot update is the `seldon-container-engine`, as it is hardcoded here:

https://github.com/SeldonIO/seldon-core/blob/01fc519a7268053b191c745c5ae6bafa0229d5ea/operator/controllers/seldondeployment_engine.go#L237-L248

https://github.com/SeldonIO/seldon-core/blob/01fc519a7268053b191c745c5ae6bafa0229d5ea/operator/controllers/seldondeployment_engine.go#L290-L301

It looks like the `docker.io/seldonio/seldon-core-executor:1.1.0` only exposes the go binary, so it's not currently possible to run an `exec` probe on it. Would this feature be useful to anyone else/ would a PR that supports this be accepted? If so, could someone please point me to the place to update the core executor image? Thanks!

**Comment by cliveseldon** [comment]
Yes, it would be good to expose this. We would need to investigate this for v2 of the CRD.

**Comment by cliveseldon** [comment]
As [v2](https://docs.seldon.io/projects/seldon-core/en/v2/) is service mesh agnostic would want to see if this is an issue in v2
